src/components/map/ha-map.ts

```diff
@@ -657,6 +659,18 @@ export class HaMap extends ReactiveElement {
     this._mapZones.forEach((marker) => map.addLayer(marker));
   }
 
+  private _getMarkerSize(computedStyles: CSSStyleDeclaration) {
+    let markerSize = 48;
+    const markerSizeVarValue =
+      computedStyles.getPropertyValue("--ha-marker-size");
+    if (markerSizeVarValue) {
+      const markerSizeVarValueSplit: string[] = markerSizeVarValue.split("px");
+      if (markerSizeVarValueSplit && markerSizeVarValueSplit.length === 2)
+        markerSize = Number(markerSizeVarValueSplit[0]);
+    }
+    return markerSize;
```

Comment:
`parseFloat` takes care of this and ignores random whitespaces
```suggestion
    const markerSizeVarValue = computedStyles.getPropertyValue("--ha-marker-size");
    const parsed = parseFloat(markerSizeVarValue);
    return Number.isNaN(parsed) ? 48 : parsed;
```